caddy.go

```diff
@@ -518,9 +518,13 @@ func startWithListenerFds(cdyfile Input, inst *Instance, restartFds map[string]r
 		}
 		if !Quiet {
 			for _, srvln := range inst.servers {
-				if !IsLoopback(srvln.listener.Addr().String()) {
-					checkFdlimit()
-					break
+				// only show FD notice if the listener is not nil.
+				// This can happen when only serving UDP or TCP
+				if srvln.listener != nil {
```

Comment:
Instead of adding a layer of nesting, how about using `if srvln.listener == nil { continue }` (with line breaks of course). Just reads a little cleaner, and I prefer it.

Reply:
yes, much better! New commit coming shortly